File: desktop/src/main/kotlin/com/mariusdev91/senin/desktop/data/WeatherRepository.kt
package com.mariusdev91.senin.desktop.data

import com.mariusdev91.senin.desktop.model.CityOption
import com.mariusdev91.senin.desktop.model.WeatherOverview

interface WeatherRepository {
    fun defaultCity(): CityOption
    fun favoriteCities(): List<CityOption>
    suspend fun searchCities(query: String): List<CityOption>
    suspend fun weatherFor(city: CityOption): WeatherOverview
}

File: senin_flutter/lib/src/data/weather_repository.dart
import '../models/city_option.dart';
import '../models/weather_models.dart';

abstract class WeatherRepository {
  CityOption defaultCity();
  List<CityOption> favoriteCities();
  Future<List<CityOption>> searchCities(String query);
  Future<WeatherOverview> weatherFor(CityOption city);
}
